readme.md

```diff
@@ -45,10 +45,42 @@ Type: `object`
 
 ##### deep
 
-Type: `boolean`<br>
+Type: `boolean | Function`<br>
 Default: `false`
 
-Recurse nested objects and objects in arrays.
+Recursing behavior for nested objects and objects in arrays.
+
+The value of `deep` can be boolean:
+
+- If `true`, it recurses all nested objects and objects in arrays.
+
+Or a function which has the following properties:
+
+- It has signature `(objectKey: string) => boolean`.
+- Returning `true` states the value related to the currently iterated `objectKey` shall be recursed (if the corresponding value is a nested object), false otherwise.
```

Comment:
This text is not in sync with https://github.com/sindresorhus/map-obj/pull/24/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R3